plugins/modules/packaging/language/cpanm.py

```diff
@@ -13,17 +13,18 @@
 module: cpanm
 short_description: Manages Perl library dependencies.
 description:
-  - Manage Perl library dependencies.
+  - Manage Perl library dependencies using cpanminus.
 options:
   name:
     type: str
     description:
-      - The name of the Perl library to install. You may use the "full distribution path", e.g.  MIYAGAWA/Plack-0.99_05.tar.gz
-    aliases: ["pkg"]
+      - The name of the Perl library to install. You may use the "full distribution path", e.g. C(MIYAGAWA/Plack-0.99_05.tar.gz).
+    aliases: [pkg, from_url]
```

Comment:
Why add a new alias here (and for `from_path` below)? We generally try to avoid aliases, so adding them needs a good reason :)

Reply:
`cpanm` allows being passed quite a range of package specifiers: name, _distribution file_ (`AUTHOR/package-x.y.tar.gz`), URL, path (as in the tar ball path) and path (directory as in the dir where the package is developed).

The rationale behind that alias (and the `from_dir` below) was to allow the user to use something semantically closer to the underlying tool. Not really a major point so, I won't fight over this.

Now I am curious, why do we try to avoid aliases?

Reply:
Having aliases usually makes the module more complicated for users: if they see an example where an option/alias they are not familiar with is used, they have to consult the documentation to find out what it does - only to see that it is an alias for something they already knew. So IMO we should try to keep aliases to a minimum.

If you think the new aliases are much better than the existing names, feel free to keep them. If they are only slightly better, I think it's better to drop them.

Reply:
nah, I removed the alias and improved the description.